src/test/sql/sql_pipeline_test.cpp

```diff
@@ -0,0 +1,385 @@
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "../base_test.hpp"
+#include "SQLParser.h"
+#include "gtest/gtest.h"
+
+#include "operators/print.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/job_task.hpp"
+#include "scheduler/node_queue_scheduler.hpp"
+#include "scheduler/topology.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_query_operator.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace opossum {
+
+class SQLPipelineTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    _table_a = load_table("src/test/tables/int_float.tbl", 2);
+    StorageManager::get().add_table("table_a", _table_a);
+
+    _table_b = load_table("src/test/tables/int_float2.tbl", 2);
+    StorageManager::get().add_table("table_b", _table_b);
+
+    _join_result = std::make_shared<Table>();
+    _join_result->add_column("a", DataType::Int);
+    _join_result->add_column("b", DataType::Float);
+    _join_result->add_column("bb", DataType::Float);
+    _join_result->append({12345, 458.7f, 456.7f});
+    _join_result->append({12345, 458.7f, 457.7f});
+  }
+
+  bool _contains_validate(const std::shared_ptr<AbstractLQPNode>& root_node) {
+    auto node = root_node;
+    while (node != nullptr) {
+      if (node->type() == LQPNodeType::Validate) return true;
+      node = node->left_child();
+    }
+    return false;
+  }
+
+  std::shared_ptr<Table> _table_a;
+  std::shared_ptr<Table> _table_b;
+  std::shared_ptr<Table> _join_result;
+
+  const std::string _select_query_a = "SELECT * FROM table_a";
+  const std::string _invalid_sql = "SELECT FROM table_a";
+  const std::string _join_query =
+      "SELECT table_a.a, table_a.b, table_b.b AS bb FROM table_a JOIN table_b ON table_a.a = table_b.a WHERE table_a.a "
+      "> 1000";
+};
+
+TEST_F(SQLPipelineTest, SimpleCreation) {
+  SQLPipeline sql_pipeline(_select_query_a);
```

Comment:
@mjendruk will think this is a function declaration, so not to confuse him, it's probably better to use curly brace hipster initializers.

Reply:
:D